truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/Node.java

```diff
@@ -517,6 +517,17 @@ public String getLanguage() {
         return "";
     }
 
+    /**
+     * Reports the execution count of a loop that is a child of this node. The optimization
+     * heuristics can use the loop count to guide compilation and inlining.
+     *
+     * @count count the number iterations to report to the runtime system
+     */
+    public final void reportLoopCount(int iterations) {
```

Comment:
@jtulach I see the concern but I have no good idea how to merge these two concepts. Not every ConditionProfile needs to be reported only loops are interesting for the runtime to compile faster.